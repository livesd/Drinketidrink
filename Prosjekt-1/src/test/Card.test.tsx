import '@testing-library/jest-dom/vitest';
import { screen, render, fireEvent } from "@testing-library/react";
import { vi } from "vitest";
import CocktailCard from "../components/card";
import type { Drink } from "../api/cocktails";

const drink: Drink = {
  idDrink: "11000",
  strDrink: "Mojito",
  strDrinkThumb: "https://example.com/mojito.jpg",
  strCategory: "Cocktail",
  strAlcoholic: "Alcoholic",
  strGlass: "Highball glass",
  strInstructions: "Mix and serve.",
};

test("renders title and toggles favorite", () => {
  const onToggle = vi.fn();

  render(
    <CocktailCard drink={drink} isFavorite={false} onToggleFavorite={onToggle} />
  );

  expect(screen.getByRole("heading", { name: /mojito/i })).toBeInTheDocument();
  const star = screen.getByRole("button", { name: /add to favorites/i });
  fireEvent.click(star);

  expect(onToggle).toHaveBeenCalledWith("11000", "Mojito");
});
