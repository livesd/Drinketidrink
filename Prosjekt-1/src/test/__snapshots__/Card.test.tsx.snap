// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component matches snapshot when favorite 1`] = `
<article
  aria-labelledby="drink-11000-title"
  class="card drink-card"
>
  <header
    class="card-header"
  >
    <h2
      id="drink-11000-title"
    >
      Mojito
    </h2>
    <button
      aria-label="Remove from favorites"
      aria-pressed="true"
      class="fav-btn"
      title="Remove from favorites"
      type="button"
    >
      ★
    </button>
  </header>
  <p
    class="meta"
  >
    <strong>
      Alcohol:
    </strong>
     
    Alcoholic
     ·
     
    <strong>
      Glass:
    </strong>
     
    Highball glass
  </p>
  <div
    class="media"
  >
    <img
      alt="Mojito"
      class="thumb"
      decoding="async"
      height="160"
      loading="lazy"
      src="https://example.com/mojito.jpg"
      width="160"
    />
    <div>
      <h3
        class="sr-only"
      >
        Ingredients
      </h3>
      <table
        aria-label="Ingredients"
        class="nutrition"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-3 oz
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              Juice of 1
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2 tsp
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-4
            </td>
          </tr>
        </tbody>
      </table>
      <h3
        class="instructions"
      >
        Instructions
      </h3>
      <p>
        Mix and serve.
      </p>
    </div>
  </div>
</article>
`;

exports[`Card component matches snapshot when not favorite 1`] = `
<article
  aria-labelledby="drink-11000-title"
  class="card drink-card"
>
  <header
    class="card-header"
  >
    <h2
      id="drink-11000-title"
    >
      Mojito
    </h2>
    <button
      aria-label="Add to favorites"
      aria-pressed="false"
      class="fav-btn"
      title="Add to favorites"
      type="button"
    >
      ☆
    </button>
  </header>
  <p
    class="meta"
  >
    <strong>
      Alcohol:
    </strong>
     
    Alcoholic
     ·
     
    <strong>
      Glass:
    </strong>
     
    Highball glass
  </p>
  <div
    class="media"
  >
    <img
      alt="Mojito"
      class="thumb"
      decoding="async"
      height="160"
      loading="lazy"
      src="https://example.com/mojito.jpg"
      width="160"
    />
    <div>
      <h3
        class="sr-only"
      >
        Ingredients
      </h3>
      <table
        aria-label="Ingredients"
        class="nutrition"
      >
        <tbody>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-3 oz
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              Juice of 1
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2 tsp
            </td>
          </tr>
          <tr>
            <th
              scope="row"
            />
            <td>
              2-4
            </td>
          </tr>
        </tbody>
      </table>
      <h3
        class="instructions"
      >
        Instructions
      </h3>
      <p>
        Mix and serve.
      </p>
    </div>
  </div>
</article>
`;
